AppDevAnnouncements/Classes/NotificationView.swift

```diff
@@ -0,0 +1,152 @@
+        let darkGray = UIColor(displayP3Red: 114/255, green: 114/255, blue: 114/255, alpha: 1)
+
+        topPortionView.backgroundColor = .white
+        addSubview(topPortionView)
+
+        bottomPortionView.backgroundColor = lightGray
+        addSubview(bottomPortionView)
+
+        dismissButton.setImage(UIImage(named: "closeIcon"), for: .normal)
+        dismissButton.addTarget(target, action: dismissFunc, for: .touchUpInside)
+        addSubview(dismissButton)
+
+        subjectLabel.textAlignment = .center
+        subjectLabel.font = UIFont.systemFont(ofSize: 24, weight: .bold)
+        subjectLabel.text = announcement.subject
+        addSubview(subjectLabel)
+
+        if let unwrappedUrl = announcement.imageUrl {
+            visualImageView.contentMode = .scaleAspectFit
+            visualImageView.image = UIImage(named: unwrappedUrl)
+            addSubview(visualImageView)
+        }
+
+        bodyLabel.numberOfLines = 0
+        bodyLabel.backgroundColor = .clear
+        bodyLabel.attributedText = Utils.attributedString(for: announcement.body)
+        addSubview(bodyLabel)
+
+        ctaButton.setTitle(announcement.ctaText, for: .normal)
+        ctaButton.setTitleColor(.white, for: .normal)
+        ctaButton.addTarget(target, action: actionFunc, for: .touchUpInside)
+        ctaButton.backgroundColor = darkGray
+        ctaButton.layer.cornerRadius = 5
+        addSubview(ctaButton)
+
+
+        setupConstraints(announcement)
+    }
+
+    struct Constants {
+        let announcement : Announcement
+        let baseTopPortionViewHeight: CGFloat = 74
+        let extraImageHeight: CGFloat = 97
+        let dismissButtonPadding: CGFloat = 12
+        let dismissButtonLength: CGFloat = 12
+        let subjectLabelTopPadding: CGFloat = 32
+        let subjectLabelHorizontalPadding: CGFloat = 24
+        let subjectLabelHeight: CGFloat = 31
+        let imageViewLength: CGFloat = 80
+        let imageViewVerticalPadding: CGFloat = 16
+        let bodyLabelTopPadding: CGFloat = 24
+        let bodyLabelHorizontalPadding: CGFloat = 16
+        let ctaButtonTopPadding: CGFloat = 16
+        let ctaButtonHorizontalPadding: CGFloat = 24
+        let ctaButtonHeight: CGFloat = 38
+        let ctaButtonBottomPadding: CGFloat = 16
+        var topPortionViewHeight: CGFloat {
+            announcement.imageUrl == nil ? baseTopPortionViewHeight: baseTopPortionViewHeight + extraImageHeight
+        }
+        var bodyLabelHeight: CGFloat {
+            let bodyLabelWidth = NotificationViewController.Constants.notificationViewWidth - bodyLabelHorizontalPadding * 2
+            return Utils.getTextHeight(for: Utils.attributedString(for: announcement.body), withConstrainedWidth: bodyLabelWidth)
+        }
+        var totalHeight: CGFloat {
+            let bottomPortionViewHeight =  bodyLabelTopPadding + bodyLabelHeight + ctaButtonTopPadding + ctaButtonHeight + ctaButtonBottomPadding
```

Comment:
couple of things:
1) Can we move this to the top of the class where the rest of the variables are
2) Can we remove `announcement` from `Constants` and also make `Constants` an enum a/ `static let` inside for the constants inside, i.e.:
```
enum Constants {
  static let baseTopPortionViewHeight: CGFloat = 74
  ...
}
```
3) Let's create functions for `topPortionViewHeight, bodyLabelHeight, totalHeight` that each take in an `announcement` as an argument, i.e.:
```
func getTopPortionViewHeight(announcement: Announcement) -> CGFloat {...} 
```